node_modules/oae-admin/tenants/js/tenants.js

```diff
@@ -311,39 +336,73 @@ define(['jquery', 'oae.core', 'jquery.jeditable'], function($, oae) {
                             'error');
-                    } else if (field === 'host') {
+                        return inlineEdit.revert;
+                    }
+                } else if (field === 'host') {
+                    if (!value || !checkValidHostname(value)) {
                         oae.api.util.notification(
                             oae.api.i18n.translate('__MSG__INVALID_HOST_NAME__', 'tenants'),
                             oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_HOST_NAME__', 'tenants'),
                             'error');
+                        return inlineEdit.revert;
                     }
-                    return this.revert;
-                } else if (field === 'host' && !checkValidHostname(value)) {
-                    oae.api.util.notification(
-                        oae.api.i18n.translate('__MSG__INVALID_HOST_NAME__', 'tenants'),
-                        oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_VALID_HOST_NAME__', 'tenants'),
-                        'error');
-                    return this.revert;
-                } else {
-                    updateTenant($inlineEdit.attr('data-alias'), field, value);
-                    // Update the link to the tenant's landing page
+                } else if (field === 'emailDomain') {
+                    if (value && !checkValidHostname(value)) {
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__INVALID_HOST_NAME__', 'tenants'),
+                            oae.api.i18n.translate('__MSG__PLEASE_ENTER_A_VALID_HOST_NAME__', 'tenants'),
+                            'error');
+                        return inlineEdit.revert;
+                    }
+                }
+
+                updateTenant($inlineEdit.attr('data-alias'), field, value, function(err) {
+                    if (err) {
+                        // Revert the field if updating failed
+                        if (inlineEdit.revert) {
+                            $inlineEdit.html(inlineEdit.revert);
+                        } else {
+                            $inlineEdit.html(placeholderContent);
+                        }
+
+                        // Revert the link to the tenant landing page if the host was updated
+                        if (field === 'host') {
+                            $('a', $inlineEdit.parent().prev()).attr('href', '//' + inlineEdit.revert);
+                        }
+                    }
+                });
+
+                // Update the link to the tenant's landing page
+                if (field === 'host') {
                     $('a', $inlineEdit.parent().prev()).attr('href', '//' + value);
-                    return value;
                 }
+
+                return value;
+
             }, {
                 'onblur': 'submit',
+                'placeholder': placeholderContent,
                 'select' : true,
```

Comment:
A lot of the changes in this file can probably be reverted when moving email domain out of the tenant list